### PR TITLE
Added new shared workflows

### PR DESCRIPTION
**Pull request by ebbi24**
Added new shared workflows:
- _build-dotnet.yml_ for building .NET Core application,
- _deploy-azure-function.yml_ for deploying applications to Azure Functions,
- _run-newman.yml_ for running Postman tests using newman.

### CODE REVIEW

File: .github/workflows/deploy-azure-function.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        required: false
+        type: string
+      bicep-file:
+        description: "Path to bicep rules file"
+        type: string
+        default: ./infrastructure/addSecurityRestriction.bicep
+      
+    secrets:
+      token:
+        required: false
```
Comment (swissarmysam) [inline .github/workflows/deploy-azure-function.yml:20]: Deploy will fail without credentials so this should maybe be `required: true` ?

**Review by swissarmysam — APPROVED** [review]
Cross-ref'd with existing pipelines and nothing stands as problematic. I have added one minor comment to review but approved.

File: .github/workflows/deploy-azure-function.yml (hunk 1 above)
Comment (ebbi24) [reply .github/workflows/deploy-azure-function.yml:20]: The reason it is not required is this workflow is being reused for testing/prod deployment. For testing deployment we don't use it though as we don't log in to Azure - that what makes it obligatory only for prod deployment

Comment (swissarmysam) [reply .github/workflows/deploy-azure-function.yml:20]: Aha, gotcha .. only used for the network rules. In which case, no comments.

**Pull request merged**